File: solutions/1.two-sum.java
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> num = new HashMap<>();

        for (int key1 = 0; key1<nums.length; key1++) {
            int val1 = nums[key1];
            int val2 = target - val1;
            if (num.containsKey(val2)) {
                int key2 = num.get(val2);
                if (key1 != key2) {
                    return new int[]{key1 , key2};
                }
            } else {
                num.put(val1, key1);
            }
        }

        return null;
    }
}
// @lc code=end

File: solutions/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num = {}

        for key1, val1 in enumerate(nums):
            val2 = target - val1
            if val2 in num:
                key2 = num[val2]
                if key1 != key2:
                    return [key1, key2]
            else:
                num[val1] = key1

        return None

# @lc code=end

